From 1d90392aad3a08fb303c2dbfd18c1cd2bc85b802 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 5 Oct 2017 02:10:54 +0200
Subject: [PATCH 1/1] efi_loader: set parent handle in efi_load_image

The parent_handle of the loaded image must be set.
Set the system table.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_boottime.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index 65939888c3..3f997fd93c 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -1266,6 +1266,8 @@ static efi_status_t EFIAPI efi_load_image(bool boot_policy,
 		return EFI_EXIT(EFI_UNSUPPORTED);
 	}
 
+	info->system_table = &systab;
+	info->parent_handle = parent_image;
 	*image_handle = info;
 
 	return EFI_EXIT(EFI_SUCCESS);
-- 
2.14.1

